src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeExperiment.groovy

```diff
@@ -14,239 +16,258 @@ import java.lang.reflect.Method
-                def dataFolder = method.invoke(null, name, relativePath, children) as DataFolder
-                folder = Optional.of(dataFolder)
-            } catch (InvocationTargetException e2) {
-                // Do nothing
-            }
-        }
-        return folder
+  /*
+   * Helper method that tries to create a DataFolder instance
+   * based on the DataFolder's different static factory create methods.
+   * As we do not know, whether a folder element is another typed folder
+   * such as FastQPassFolder or a named folder such as Fast5Folder, we have to
+   * try and fail.
+   */
+
+  private static Optional<DataFolder> tryToCreateDataFolder(Method method,
+                                                            String name,
+                                                            String relativePath,
+                                                            List children) {
+    Optional<DataFolder> folder = Optional.empty()
+    try {
+      // Try typed folder
+      def dataFolder = method.invoke(null, relativePath, children) as DataFolder
+      folder = Optional.of(dataFolder)
+    } catch (InvocationTargetException e) {
+      // Do nothing
+    } catch (IllegalArgumentException e) {
+      try {
+        // Try named folder
+        def dataFolder = method.invoke(null, name, relativePath, children) as DataFolder
+        folder = Optional.of(dataFolder)
+      } catch (InvocationTargetException e2) {
+        // Do nothing
+      }
     }
+    return folder
+  }
 
-    /*
-     * Helper method that parses the children of a folder.
-     */
-
-    private static List parseChildren(List<Map> children) {
-        def parsedChildren = []
-        children.each { Map unknownChild ->
-            try {
-                def child = parseFile(unknownChild)
-                parsedChildren.add(child)
-            } catch (IllegalArgumentException e) {
-                // We do not capture the second parse call, as we want to fail the parsing at this point.
-                // This means that we ultimately found a child of unknown type, which should
-                // break the parsing.
-                def child = parseFolder(unknownChild)
-                parsedChildren.add(child)
-            }
-        }
-        return parsedChildren
+  /*
+   * Helper method that parses the children of a folder.
+   */
+
+  private static List parseChildren(List<Map> children) {
+    def parsedChildren = []
+    children.each { Map unknownChild ->
+      if (unknownChild.get(FILEKEY)) {
+        def child = parseFile(unknownChild)
+        parsedChildren.add(child)
+      }
+      if (unknownChild.get(FOLDERKEY)) {
+        // We do not capture the second parse call, as we want to fail the parsing at this point.
+        // This means that we ultimately found a child of unknown type, which should
+        // break the parsing.
+        def child = parseFolder(unknownChild)
+        parsedChildren.add(child)
+      }
```

Comment:
reminds me of the measurement item parsing. Please refer to my comments there.

Reply:
Will do 🥳